plugins/button/plugin.js

```diff
@@ -337,14 +343,11 @@
 				element.setState( state, 'cke_button' );
 				element.setAttribute( 'aria-disabled', state == CKEDITOR.TRISTATE_DISABLED );
 
-				if ( !this.hasArrow ) {
-					// Note: aria-pressed attribute should not be added to menuButton instances. (https://dev.ckeditor.com/ticket/11331)
-					if ( state === CKEDITOR.TRISTATE_ON ) {
-						element.setAttribute( 'aria-pressed', true );
-					} else {
-						element.removeAttribute( 'aria-pressed' );
-					}
-				} else {
+				if ( this.isToggle && !this.hasArrow ) {
+					// Note: aria-pressed attribute should not be added to menuButton instances. (https://dev.ckeditor.com/ticket/11331).
+					// Do not remove the attribute, set its value (#2444).
+					element.setAttribute( 'aria-pressed', state === CKEDITOR.TRISTATE_ON );
```

Comment:
I'm not sure if it's not introducing a breaking change. If someone is already using a button (3rd party plugin) without an arrow, it supports the `aria-pressed` feature. Maybe we should instead keep the previous logic for the non-defined `isToggle` value?

Reply:
Uh, alright, that wouldn't make much sense as we will be telling that minimalize is active.

Reply:
Looks like it was setting `aria-pressed` for `!this.hasArrow` check? Although I don't see it to be working correctly before your changes on VoiceOver despite having `aria-pressed="true"` for toggled button - not sure why.

Reply:
Because toggle buttons are identified by the presence of the `[aria-pressed]` attribute. If the attribute is not present, it's considered a normal button. So adding that attribute only for the `true` value causes that the button is a toggle one only when it's pressed (in Chrome it isn't at all).